pytest_fixtures/component/provision_pxe.py

```diff
@@ -189,39 +191,41 @@ def module_provisioning_sat(
     # we might need to set up Sat's DNS server as the primary one on the Sat host
     provisioning_upstream_dns_primary = (
         broker_data_out.provisioning_upstream_dns
-        if settings.server.is_ipv6
+        if sat_ipv6
         else broker_data_out.provisioning_upstream_dns.pop()
     )  # There should always be at least one upstream DNS
     provisioning_upstream_dns_secondary = (
         broker_data_out.provisioning_upstream_dns.pop()
-        if len(broker_data_out.provisioning_upstream_dns) and not settings.server.is_ipv6
+        if len(broker_data_out.provisioning_upstream_dns) and not sat_ipv6
         else None
     )
 
     domain = sat.api.Domain(
         location=[module_location],
         organization=[module_sca_manifest_org],
-        dns=None if settings.server.is_ipv6 else module_provisioning_capsule.id,
+        dns=None if sat_ipv6 else module_provisioning_capsule.id,
         name=provisioning_domain_name,
     ).create()
 
     subnet = sat.api.Subnet(
         location=[module_location],
         organization=[module_sca_manifest_org],
         network=str(provisioning_network.network_address),
-        network_type='IPv6' if settings.server.is_ipv6 else 'IPv4',
+        network_type=NetworkType.IPV4.formatted
+        if sat.network_type == NetworkType.IPV4
+        else NetworkType.IPV6.formatted,
```

Comment:
I'd also say that the `network_type` should be derived from `provisioning_network`, not `sat.network_type`. So really:
```suggestion
        network_type='IPv6' if provisioning_network.version == 6 else 'IPv4',
```